inspirehep/testlib/api/base_resource.py

```diff
@@ -34,3 +34,12 @@ def __repr__(self):
             if not item[0].startswith('_')
         )
         return '{classname}({args})'.format(classname=classname, args=args)
+
+    def __eq__(self, other):
+        for this_prop, other_prop in zip(dir(self), dir(other)):
+            if this_prop != other_prop:
+                return False
+
+        return len(dir(self)) == len(dir(other))
+
+        return True
```

Comment:
you never get to this line, as you have an unconditional `return` just above.

Reply:
Too late :/, opened another pr for it